probe-rs/src/probe.rs

```diff
@@ -367,7 +368,7 @@ impl Probe {
 
     /// Selects the transport protocol to be used by the debug probe.
     pub fn select_protocol(&mut self, protocol: WireProtocol) -> Result<(), DebugProbeError> {
-        if !self.attached {
+        if self.attached {
```

Comment:
this seems wrong. If the target is already attached, the protocol cannot be changed. What am I missing?